### PR TITLE
feat: flush & flushSync

### PR DESCRIPTION
**Pull request by davidmarkclements**
Due to massive amounts of changes this re-implements #423 - however I started with @ronag s
commit and built on there to ensure @ronag is also listed as a contributor for this

The docs for this will go in as part of the docs pr, #447

### CODE REVIEW

**Comment by davidmarkclements** [comment]
strange - ci is red – tests are passing locally - need to look into this but out of time for tonight

File: pino.js [hunk 1]
```diff
@@ -82,10 +82,13 @@ function pino (...args) {
   const time = (timestamp instanceof Function)
     ? timestamp : (timestamp ? epochTime : nullTime)
 
+  const extreme = stream.minLength === 4096
```
Comment (mcollina) [inline pino.js]: I think this make sense if stream is a SonicBoom instance instead.

Comment (davidmarkclements) [reply pino.js]: then there's no need for noop - which is a little less explicit  - what if both? 

```
const extreme = stream instanceof SonicBoom && stream.minLength === 4096
```

Comment (mcollina) [reply pino.js]: right, we need to check both

**Comment by davidmarkclements** [comment]
supsceded by #454

**Pull request closed without merging**